src/main/java/org/hoverla/bibernate/connectionpool/util/DriverDataSource.java

```diff
@@ -0,0 +1,145 @@
+package org.hoverla.bibernate.connectionpool.util;
+
+import lombok.extern.slf4j.Slf4j;
+
+import javax.sql.DataSource;
+import java.io.PrintWriter;
+import java.sql.*;
+import java.util.Properties;
+
+/**
+ * Utility data source to get low-level connections to filling the pool.
+ */
+@Slf4j
+public class DriverDataSource implements DataSource {
+
+    private static final String PASSWORD = "password";
+    private static final String USER = "user";
+
+    private final String jdbcUrl;
+    private final Properties driverProperties;
+    private Driver driver;
+
+    public DriverDataSource(String jdbcUrl, String driverClassName, String username, String password) {
+        this.jdbcUrl = jdbcUrl;
+        this.driverProperties = new Properties();
+
+        if (username != null) {
```

Comment:
Maybe better do the `Configuration` validation in `initializeDataSource` method? This way we don't need these checks here.